src/SASViyaApiClient.ts

```diff
@@ -1290,6 +1303,47 @@ export class SASViyaApiClient {
     )
   }
 
+  /**
+   * Lists a children folders for given Viya folder.
+   * @param sourceFolder - the full path (eg `/Public/example/myFolder`) or URI of the source folder listed. Providing URI instead of path will save one extra request.
+   * @param accessToken - an access token for authorizing the request.
+   */
+  public async listFolder(sourceFolder: string, accessToken?: string) {
+    // checks if 'sourceFolder' is already a URI
+    const sourceFolderUri = isUri(sourceFolder)
+      ? sourceFolder
+      : await this.getFolderUri(sourceFolder, accessToken)
+
+    const requestInfo = {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + accessToken
+      }
+    }
+
+    const url = sourceFolderUri
+
+    const { result: members } = await this.request<{ items: any[] }>(
+      `${this.serverUrl}${url}/members`,
+      requestInfo
+    ).catch((err) => {
+      if (err.code && err.code === 'ENOTFOUND') {
+        const notFoundError = {
+          body: JSON.stringify({
+            message: `Folder '${sourceFolder.split('/').pop()}' was not found.`
+          })
+        }
+
+        throw notFoundError
+      }
+
+      throw err
```

Comment:
It would be nice to prefix `err` with some custom message. You can use `import { prefixMessage } from '@sasjs/utils/error'`.